pydax/_dataset.py

```diff
@@ -161,6 +161,36 @@ def is_downloaded(self) -> bool:
         return self._data_dir.is_dir() and len(os.listdir(self._data_dir)) > 0
 
 
+def handle_name_version_params(func: Callable) -> Callable:
```

Comment:
This is something we are craving after. Could you add a leading underscore to the function name? In fact, this might be more useful if you split this to `name` and `version`, and we can decorate either or them or both.

Reply:
I'm having a hard time figuring out how to split this into two separate decorators. Is there a way to create a decorator that modifies a function's argument without then executing the function?

Reply:
I think I can create one decorator that takes has parameters that control what arguments it should modify e.g. 
```python
import functools
def _params_to_modify(name=True, version=True):
    def modify_params(func):
        @functools.wraps(func)
        def wrapper(*args, **kwargs):
            if name:
                ...
            if version:
                ...
            value = func(name=name, version=version, *args, **kwargs)
            return value
        return wrapper
```

Reply:
Just have this style of function twice for `name` and `version` respectively (e.g., move all `version` related code to the other function). Then, when using them, just stack these decorators. Does this answer your question?

Reply:
Not really, my problem is that decorators seem like they have to run the function they decorate, so if we have two separate decorators, then wouldn't that run the decorated function twice in a row? 

```python
def decorator(func):
	def wrapper_decorator(*args, **kwargs):
		# Do something before
		value = func(*args, **kwargs)  # <-- runs the function
		# Do something after
		return value
	return wrapper_decorator
```

Reply:
What do you think about using the method I posted [above](https://github.com/CODAIT/pydax/pull/30#discussion_r535631016)?

Reply:
You don't run the decorated function in the decorator, because you return the `wrapper_decorator`, not running it. After expansion, the return value would look like (assuming 2 decorators):

```python
def wrapper_decorator_outer(*args, **kwargs):
    wrapper_decorator_inner(*args, **kwargs)

def wrapper_decorator_inner(*args, **kwargs):
    value = func(*args, **kwargs)
    return value
```